src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -379,35 +379,15 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
         Assert.notNull(domainClass, "domainClass should not be null.");
         Assert.hasText(collectionName, "collection should not be null, empty or only whitespaces");
 
-        if (query.getSort().isSorted()) { // avoiding unnecessary query with DocumentCollection
-            query.validateSort(domainClass, isCollectionSupportSortByString(getDocCollection(collectionName)));
-        }
+        QueryValidator.validateQuery(query, domainClass, getDocCollection(collectionName));
```

Comment:
`getDocCollection` will query to server, right ?  Can we suspend this query later ?

Reply:
Yes, I'll think about this later.